guide/additional-features/cooldowns.md

```diff
@@ -0,0 +1,94 @@
+
+This section assumes you followed the [Command Handling](/guide/creating-your-bot/command-handling.md) part.
+
+First, add a cooldown key to one of your commands (we use the ping command here) exports. This determines how long the user will have to wait (in seconds) before using this specific command again.
+
+```js {3}
+const { SlashCommandBuilder } = require('discord.js');
+
+module.exports = {
+	cooldown: 5,
+	data: new SlashCommandBuilder()
+		.setName('ping')
+		.setDescription('Replies with Pong!'),
+	async execute(interaction) {
+		// ...
+	},
+};
+```
+
+In your main file, initialize an empty [Collection](/guide/additional-info//collections.md) which you can then fill later when commands are used:
+
+```js {2}
+client.cooldowns = new Collection();
+```
+
+The keys will be the command names, and the values will be Collections associating the user id (key) to the last time (value) this specific user used this specific command. Overall the logical path to get a specific user's last usage of a specific command will be `cooldowns > command > user > timestamp`.
+
+In your main file, add the following code:
+
+```js {1,3-5,7-9,11-13}
+const { cooldowns } = client;
+
+if (!cooldowns.has(command.data.name)) {
+	cooldowns.set(command.data.name, new Collection());
+}
+
+const now = Date.now();
+const timestamps = cooldowns.get(command.data.name);
+const defaultCooldownDuration = 3;
+const cooldownAmount = (command.cooldown || defaultCooldownDuration) * 1000;
+
+if (timestamps.has(interaction.user.id)) {
+	// ...
+}
+
+try {
+	// ...
+} catch (error) {
+	// ...
+}
+```
+
+You check if the `cooldowns` Collection already has an entry for the command being used right now. If this is not the case, add a new entry, where the value is initialized as an empty Collection. Next, 3 variables are created:
+
+1. `now`: The current timestamp.
+2. `timestamps`: A reference to the Collection of user-ID and timestamp key/value pairs for the triggered command.
+3. `cooldownAmount`: The specified cooldown from the command file, converted to milliseconds for straightforward calculation. If none is specified, this defaults to three seconds.
+
+If the user has already used this command in this session, get the timestamp, calculate the expiration time and inform the user of the amount of time they need to wait before using this command again. Note that you use a `return` statement here, causing the code below this snippet only to execute if the interaction user has not used this command in this session or the wait has already expired.
+
+Continuing with your current setup, this is the complete `if` statement:
+
+```js {2,4-7}
+if (timestamps.has(interaction.user.id)) {
+	const expirationTime = timestamps.get(interaction.user.id) + cooldownAmount;
+
+	if (now < expirationTime) {
+		const timeLeft = Math.round(expirationTime / 1000);
```

Comment:
maybe raname to expiritationTimeInSeconds

Reply:
a comment would be better maybe

Reply:
yeah

Reply:
timeLeft seems self explanatory

Reply:
> timeLeft seems self explanatory

for old when it were seconds left , ok
but NOT NOW when its timestamp in seconds